include/text.php

```diff
@@ -1196,6 +1197,16 @@ function put_item_in_cache(&$item, $update = false)
 		$item["rendered-html"] = prepare_text($item["body"]);
 		$item["rendered-hash"] = hash("md5", $item["body"]);
 
+		// Force an update if the generated values differ from the existing ones
+		if ($rendered_hash != $item["rendered-hash"]) {
+			$update = true;
+		}
+
+		// Only compare the HTML when we forcefully ignore the cache
+		if (Config::get("system", "ignore_cache") && ($rendered_html != $item["rendered-html"])) {
```

Comment:
Wait, if we ignore the item cache, we shouldn’t store it, right?

Reply:
If we don't ignore the cache, we only need to store when the hash differs.

Reply:
Not only, since you had an issue with updating the phpBB library and an item that had the same body ash, but not the same HTML content.